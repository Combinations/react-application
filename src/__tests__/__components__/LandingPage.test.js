import React from 'react';
import LandingPage from '../../components/Landing';
import {render, cleanup} from 'react-testing-library'
import  {BrowserRouter as Router} from 'react-router-dom'
import 'jest-dom/extend-expect'

afterEach(cleanup)

const renderComponent = () => render(
  <Router>
    <LandingPage/>
  </Router>
);

it('always renders content', () => {
  const {container} = renderComponent()
  expect(container.firstChild).toMatchSnapshot() //ensure that the page's copy/layout has not changed.
})

it('should render its children components', ()=> {
  const {getByTestId} = renderComponent()

  //assert that the two children components are in the dom
  expect(getByTestId('AgeRestrictionAlert')).toBeInTheDocument()
  expect(getByTestId('LandingFooter')).toBeInTheDocument()
})

it('should have a background image in the hero section', () => {
  const {getByTestId} = renderComponent()
  expect(getByTestId('hero')).toHaveStyle(`background-image: url(meadow.jpg)`)
})

it('should have two sign up buttons that link to the sign up page', () => {
  const {queryAllByText} = renderComponent()
  const signUpButtons = queryAllByText('Sign up')

  //ensure that the buttons are visible and are in the DOM
  expect(signUpButtons).toHaveLength(2)
  expect(signUpButtons[0]).toBeVisible()
  expect(signUpButtons[1]).toBeVisible()

  //ensure that the buttons link to the correct page
  expect(signUpButtons[0].href).toEqual(`${base_url}/signup`)
  expect(signUpButtons[1].href).toEqual(`${base_url}/signup`)
})

it('should have two sign in buttons that link to the sign in page', () => {
  const {queryAllByText} = renderComponent()
  const signInButtons = queryAllByText('Sign in')

  //ensure that the buttons are visible and are in the DOM
  expect(signInButtons).toHaveLength(2)
  expect(signInButtons[0]).toBeVisible()
  expect(signInButtons[1]).toBeVisible()

  //ensure that the buttons link to the correct page
  expect(signInButtons[0].href).toEqual(`${base_url}/signin`)
  expect(signInButtons[1].href).toEqual(`${base_url}/signin`)
})

