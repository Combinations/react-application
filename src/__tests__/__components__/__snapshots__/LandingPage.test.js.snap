// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`always renders content 1`] = `
<div>
  <div
    data-testid="AgeRestrictionAlert"
    id="modal1"
    class="modal"
    tabindex="0"
  >
    <div
      class="modal-content"
    >
      <h4>
        Age Confirmation
      </h4>
      <p>
        Are you over 19 years of age?
      </p>
    </div>
    <div
      class="modal-footer"
    >
      <a
        class="modal-action modal-close waves-effect btn-flat"
        href="/age-restriction"
      >
        No
      </a>
      <a
        href="#!"
        class="modal-action modal-close waves-effect btn-flat"
      >
        Yes
      </a>
    </div>
  </div>
  <section
    id="hero"
    data-testid="hero"
    style="background-image: url(meadow.jpg);"
  >
    <div
      class="container valign-wrapper jc-center"
    >
      <div
        class="valign center-align white-text"
      >
        <p
          class="flowtext hide-on-small-only"
        >
          The modern dispensary 
        </p>
        <h3>
          Browse, Select, Order, Track, Enjoy 
        </h3>
        <p
          class="big"
        >
          We produce your medicine using cutting edge biology, chemistry and software
          <br />
          Order now and enjoy a truly superior experience
        </p>
        <a
          class="btn m-r-16 grey darken-3"
          href="/signup"
        >
          Sign up
        </a>
        <a
          class="btn m-r-16 grey darken-3"
          href="/signin"
        >
          Sign in
        </a>
      </div>
    </div>
  </section>
  <section
    class="grey lighten-3"
  >
    <div
      class="container"
    >
      <div
        class="col m6 center-align"
      >
        <h2>
          How it works
        </h2>
      </div>
      <div
        class="row"
      >
        <div
          class="col m3"
        >
          <h5>
            <i
              class="material-icons"
            >
              looks_one
            </i>
            Browse
          </h5>
          <p>
            Browse hundreds of flowers, strains, concentrates and accessories from our constantly evolving offerings.
          </p>
        </div>
        <div
          class="col m3"
        >
          <h5>
            <i
              class="material-icons"
            >
              looks_two
            </i>
            Select
          </h5>
          <p>
            Select the items you want. Our intuitive user interface will guide you. It's simple, quick and easy.
          </p>
        </div>
        <div
          class="col m3"
        >
          <h5>
            <i
              class="material-icons"
            >
              looks_3
            </i>
            Order
          </h5>
          <p>
            Order your selected items through our secure checkout. Our checkout system uses industry standard encryption (HTTPS). 
          </p>
        </div>
        <div
          class="col m3"
        >
          <h5>
            <i
              class="material-icons"
            >
              looks_4
            </i>
            Track
          </h5>
          <p>
            Track your orders. We've built you an amazing tracker so that you'll be informed on the status of your order.
          </p>
        </div>
      </div>
    </div>
  </section>
  <section>
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col m6 offset-m3 center"
        >
          <h2>
            Our Approach 
          </h2>
          <p>
            By utilizing modern biology, chemistry, and software we are able to optimize the seed-to-patient process. This enables us to bring you extremely high-quality product at reasonable costs.
          </p>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col m3"
        >
          <h5>
            <i
              class="material-icons"
            >
              group
            </i>
            Team
          </h5>
          <p>
            Our team consists of engineering, law, chemistry and botany professionals. We have experience in a range of industries. We are educated from academic insitutions from around the world. We work hard everyday in order to bring you high-quality medical grade cannabis.
          </p>
        </div>
        <div
          class="col m3"
        >
          <h5>
            <i
              class="material-icons"
            >
              account_balance
            </i>
            Facilities
          </h5>
          <p>
            Our custom facilities have completely revolutionized the way natural medicinal cannabis is grown. Our facilities are built with a strong emphasis on automation. We use machine learning in order to care for each plant on an individual basis.
          </p>
        </div>
        <div
          class="col m3"
        >
          <h5>
            <i
              class="material-icons"
            >
              build
            </i>
            Online
          </h5>
          <p>
            We are an online first dispensary. This allows us to avoid the costs of physical stores and to focus our efforts on improving the quality of our products and facilities.
          </p>
        </div>
        <div
          class="col m3"
        >
          <h5>
            <i
              class="material-icons"
            >
              mood
            </i>
            Partnerships
          </h5>
          <p>
            We have great partnerships with government, academic insituitions, and hospitals. We are exicited to work with any organization that wishes to push forward the understanding of medicinal cannabis. 
          </p>
        </div>
      </div>
    </div>
  </section>
  <section
    class="grey lighten-3"
  >
    <div
      class="container valign-wrapper jc-center"
    >
      <div
        class="valign center-align"
      >
        <h2>
          Ready to get started? 
        </h2>
        <p
          class="big"
        >
          Sign in to browse, or sign up to create an account 
        </p>
        <a
          class="btn m-r-16 grey darken-3"
          href="/signup"
        >
          Sign up
        </a>
        <a
          class="btn m-r-16 grey darken-3"
          href="/signin"
        >
          Sign in
        </a>
      </div>
    </div>
  </section>
  <footer
    data-testid="LandingFooter"
    class="page-footer grey darken-3"
  >
    <div
      class="container"
    >
      <div
        class="row center social"
      >
        <div
          class="col s3"
        >
          <a
            target="_blank"
            href="/https://Facebook.com"
          >
            <i>
              <img
                src="facebook-box.png"
                alt="facebook social link"
              />
            </i>
          </a>
        </div>
        <div
          class="col s3"
        >
          <a
            target="_blank"
            href="/https://Instagram.com"
          >
            <i>
              <img
                src="instagram.png"
                alt="instagram social link"
              />
            </i>
          </a>
        </div>
        <div
          class="col s3"
        >
          <a
            target="_blank"
            href="/https://twitter.com"
          >
            <i>
              <img
                src="twitter.png"
                alt="twitter social link"
              />
            </i>
          </a>
        </div>
        <div
          class="col s3"
        >
          <a
            target="_blank"
            href="/https://snapchat.com"
          >
            <i>
              <img
                src="snapchat.png"
                alt="snapchat social link"
              />
            </i>
          </a>
        </div>
      </div>
    </div>
    <section
      id="footer"
      class="grey darken-3"
    >
      <div
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="col s12 center-align grey-text"
          >
            Â© 2018 All Rights Reserved Terms of Use and Privacy Policy
          </div>
        </div>
      </div>
    </section>
  </footer>
</div>
`;
